sisyphus/job.py

```diff
@@ -684,8 +684,15 @@ def _sis_all_path_available(self):
                 return False
         return True
 
-    def _sis_runnable(self):
+    def _sis_runnable(self, *, return_when_finished: Optional[bool] = True):
         """True if all inputs are available, also checks if new inputs are requested"""
+        if return_when_finished is not None:
```

Comment:
Shouldn't this be `if return_when_finishes is True`? Otherwise putting False would also move into this branch, which would be unexpected to me.

Reply:
If you want to do a boolean check, you would just do `if return_when_finishes: ...` (never do `... is True`, that is not standard Python code style).

But the type here intentionally is not just `bool` but `Optional[bool]`. The meaning of `None` is that it should not use it. If it is not None, it should return it. Both False/True can be valid return values.

I could maybe add some docstring for this, like this?
```
:param return_when_finishes: when set, it immediately returns this value when it is finished,
    instead of checking the inputs
```
(Actually, the name maybe should be better `return_when_finished`.)

Note, I made this a possibility to return both True and False for the case it is finished, because it's not clear to me that in all possible cases always returning True is correct, because when this option is used, we don't actually check whether the job is really runnable.

Reply:
>never do ... is True, that is not standard Python code style

I know, but I like not giving the use the option to cause weird behaviour by passing something wrong (like `return_when_finished=5`).

But I see your point, I missed that the value itself is returned, then I agree with your variant.

Reply:
> to cause weird behaviour by passing something wrong

If you want to guard against that, I would just do an explicit type check, like:
```python
assert return_when_finished is None or isinstance(return_when_finished, bool)
```